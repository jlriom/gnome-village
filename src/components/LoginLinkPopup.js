import React, { useState } from "react";
import { Nav } from "react-bootstrap";
import LoginPage from "../containers/auth/LoginPage";

const LoginLinkPopup = () => {
	const [show, setShow] = useState(false);
	return (
		<>
			<Nav.Link onClick={() => setShow(true)}>Login</Nav.Link>
			<LoginPage show={show} handleClose={() => setShow(false)}></LoginPage>
		</>
	);
};

export default LoginLinkPopup;
