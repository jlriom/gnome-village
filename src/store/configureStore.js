import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
