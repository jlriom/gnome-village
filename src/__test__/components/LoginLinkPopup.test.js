import React from "react";
import LoginLinkPopup from "../../components/LoginLinkPopup";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import { Provider } from "react-redux";

import configureStore from "../../store/configureStore";
const store = configureStore();

describe("<LoginLinkPopup />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <LoginLinkPopup />
                </MemoryRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});
