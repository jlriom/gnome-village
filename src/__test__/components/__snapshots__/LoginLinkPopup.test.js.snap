// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginLinkPopup /> renders ok 1`] = `
Array [
  <a
    className="nav-link"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    Login
  </a>,
  <div />,
]
`;
